modules/launcher/src/main/java/org/mule/module/launcher/DefaultAppBloodhound.java

```diff
@@ -67,35 +67,16 @@ public DefaultAppBloodhound()
     public ApplicationDescriptor fetch(String appName) throws IOException
     {
         final File appsDir = MuleContainerBootstrapUtils.getMuleAppsDir();
-        File appDir = new File(appsDir, appName);
-        if (!appDir.exists())
-        {
-            throw new MuleRuntimeException(
-                    MessageFactory.createStaticMessage(
-                            String.format("Application directory does not exist: '%s'", appDir)));
-        }
-        // list mule-deploy.* files
-        @SuppressWarnings("unchecked")
-        Collection<File> deployFiles = FileUtils.listFiles(appDir, new WildcardFileFilter("mule-deploy.*"), null);
-        if (deployFiles.size() > 1)
-        {
-            // TODO need some kind of an InvalidAppFormatException
-            throw new MuleRuntimeException(
-                    MessageFactory.createStaticMessage(
-                            String.format("More than one mule-deploy descriptors found in application '%s'", appName)));
-        }
-
+        final File descriptorFile = ArtifactFactoryUtils.getDeploymentFile(appsDir, appName);
```

Comment:
You can use MuleFolderUtils#getAppFolder instead of passing appsDir and and name to getDeploymentFile method